inc/dashboardfilters.class.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+/**
+ * -------------------------------------------------------------------------
+ * Credit plugin for GLPI
+ * -------------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of Credit.
+ *
+ * Credit is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Credit is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Credit. If not, see <http://www.gnu.org/licenses/>.
+ * -------------------------------------------------------------------------
+ * @author    François Legastelois
+ * @copyright Copyright (C) 2017-2023 by Credit plugin team.
+ * @license   GPLv3 https://www.gnu.org/licenses/gpl-3.0.html
+ * @link      https://github.com/pluginsGLPI/credit
+ * -------------------------------------------------------------------------
+ */
+
+use Glpi\Dashboard\Filters\AbstractFilter;
+use PluginCreditEntity;
+
+class PluginCreditDashboardFilters extends AbstractFilter
+{
+    const FIELD = "id";
+
+    public static function getName(): string
+    {
+        return __("Crédit");
+    }
+
+    public static function getId(): string
+    {
+        return "credit";
+    }
+
+    public static function canBeApplied(string $table): bool
+    {
+        /** @var \DBmysql $DB */
+        global $DB;
+        return $DB->fieldExists($table, self::FIELD);
```

Comment:
Is always applied on all table (because `id` field always exist in table)

Reply:
I just updated the FIELD constant to "plugin_credit_types_id" so that the filter only applies to new cards linked to credits
[https://github.com/pluginsGLPI/credit/pull/152/commits/3dde920d3392d42c7ab7699ea2ec57ddb7c7b8b1](https://github.com/pluginsGLPI/credit/pull/152/commits/3dde920d3392d42c7ab7699ea2ec57ddb7c7b8b1)